src/libtexture/imagecache.cpp

```diff
@@ -467,6 +470,7 @@ ImageCacheFile::open (ImageCachePerThreadInfo *thread_info)
     // Since each subimage can potentially have its own mipmap levels,
     // keep track of the highest level discovered
```

Comment:
I'm not sure I follow this part.

We only count the size of the largest subimage? Why not count the size of the subimage we actually read?

Does the same logic make sense for the filesize?